src/Runtime/Factors/StringFormatOp.cs

```diff
@@ -754,15 +756,38 @@ public static double LogEvidenceRatio(string str, StringDistribution format, ILi
         /// <returns>The generated array of argument names.</returns>
         private static string[] GetArgumentNames(int argCount)
```

Comment:
Change return value to ReadOnlyArray<>, this way we will avoid unneccessary copying